### PR TITLE
Impr/facebook models

### PR DESCRIPTION
**Pull request by sherali-arbisoft**

### CODE REVIEW

File: digital_benchmark/facebook_benchmark/data_parser.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from django.conf import settings
+from django.utils import timezone
+
+from . import models
+
+class FacebookDataParser:
+
+    def parse_page_details_and_insights(self, page_response, page_insights_response):
+        page = models.Page()
```
Comment (ehmadzubair) [inline digital_benchmark/facebook_benchmark/data_parser.py]: Import Page rather than models

File: digital_benchmark/facebook_benchmark/models.py [hunk 2]
```diff
@@ -10,7 +10,7 @@ class Meta:
         abstract = True
 
 class CreateUpdateMixin(models.Model):
-    created_at = models.DateTimeField()
+    created_at = models.DateTimeField(default=timezone.now)
```
Comment (ehmadzubair) [inline digital_benchmark/facebook_benchmark/models.py]:
```created_at = models.DateTimeField(auto_now_add=True, editable=False)
last_updated_at = models.DateTimeField(auto_now=True, editable=False)```

File: digital_benchmark/facebook_benchmark/data_parser.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        page.created_at = timezone.now()
```
Comment (ehmadzubair) [inline digital_benchmark/facebook_benchmark/data_parser.py]: No need to do this, check the CreateUpdateMixin model comment

File: digital_benchmark/facebook_benchmark/data_parser.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        page.updated_at = timezone.now()
+        for key in settings.FACEBOOK_DEFAULT_FIELDS_FOR_PAGE:
+            setattr(page, key, page_response.get(key, ''))
+        page.id = ''
```
Comment (ehmadzubair) [inline digital_benchmark/facebook_benchmark/data_parser.py]: No need to do this. Django automatically hanldes this

File: digital_benchmark/facebook_benchmark/data_parser.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        page.page_id = page_response['id']
+        page.engagement = page_response['engagement']['count']
```
Comment (ehmadzubair) [inline digital_benchmark/facebook_benchmark/data_parser.py]: rename this field to `num_engagements`

File: digital_benchmark/facebook_benchmark/data_parser.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        for key in settings.FACEBOOK_DEFAULT_METRICES_FOR_PAGE_INSIGHTS:
+            items = [item for item in page_insights_response if item.get('name', '') == key]
+            if len(items) > 0:
+                setattr(page, key, items[0]['values'][0]['value'])
```
Comment (ehmadzubair) [inline digital_benchmark/facebook_benchmark/data_parser.py]: Why are we hardcoding the index?

File: digital_benchmark/facebook_benchmark/data_provider.py [hunk 7]
```diff
@@ -13,24 +13,49 @@ def get_page_details(self, fields=''):
...
+    def get_next(self, next_url):
+        return requests.get(next_url).json()
+    
+    def append_reactions(self, post):
+        reactions = post['reactions']
+        while 'paging' in reactions and 'next' in reactions['paging']:
+            reactions = self.get_next(reactions['paging']['next'])
+            post['reactions']['data'] += reactions['data']
+    
+    def append_comments(self, post):
```
Comment (ehmadzubair) [inline digital_benchmark/facebook_benchmark/data_provider.py]: prefix the private methods with a leading underscore... `_append_comments`

**Comment by ehmadzubair** [comment]
Some comments. The rest looks good

File: digital_benchmark/facebook_benchmark/data_parser.py (hunk 6 above)
Comment (sherali-arbisoft) [reply digital_benchmark/facebook_benchmark/data_parser.py]: this is old code and has been updated on this PR "Feat/facebook data parser #9"

**Pull request merged**